cartographer/pose_graph/constraint/constraint.h

```diff
@@ -45,8 +48,13 @@ class Constraint {
  protected:
   virtual proto::CostFunction ToCostFunctionProto() const = 0;
 
+  const std::unique_ptr<ceres::LossFunction>& ceres_loss() const {
```

Comment:
Returning a std::unique_ptr<>& looks weird. Remove a pointer instead? (To me returning a unique_ptr<> always indicates ownership transfer)

Reply:
I meant 'return a pointer instead'